web_programming/get_links_of_url.py

```diff
@@ -0,0 +1,60 @@
+import re
+import urllib.parse
+
+import requests
+from bs4 import BeautifulSoup
+
+
+def is_valid_url(url):
+    """
+    Function Valid the url
+    """
+
+    regex = re.compile(
+        r"^(?:http|ftp)s?://"
+        r"(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)"
+        r"+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|"
+        r"localhost|"
+        r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"
+        r"(?::\d+)?"
+        r"(?:/?|[/?]\S+)$",
+        re.IGNORECASE,
+    )
+    return re.match(regex, url) is not None
```

Comment:
Please use atleast 10 ***different*** types of ***correct*** and 10 ***different*** types of ***incorrect*** URLs to test this function as mentioned previously. Example:

```python
def is_valid_url(url):
    """
    >>> is_valid_url(https://www.github.com)
    True
    ...
    """